server/src/Persistence.hs

```diff
@@ -173,12 +173,29 @@ recoverJournal pval = for_ (pvJournal pval) $ \journalHandle -> ExceptT $ fmap f
     failedRecoveryMsg err line = "Failed to recover journal entry "
       <> Text.pack (show line) <> ": " <> Text.pack err
 
--- | Read and decode the data file
-readData :: FilePath -> ExceptT String IO Store.Value
-readData filePath = ExceptT $ do
-  eitherEncodedValue <- try $ withFile filePath ReadMode SBS.hGetContents
-  case (eitherEncodedValue :: Either SomeException SBS.ByteString) of
-    Left exc -> pure $ Left $ "Failed to read the data from disk: " ++ show exc
+-- | Read and decode the data file from disk
+readData :: FilePath -> Logger -> ExceptT String IO Store.Value
+readData filePath logger = ExceptT $ do
+  eitherEncodedValue <- tryIOError $ withFile filePath ReadMode SBS.hGetContents
```

Comment:
It should be possible to replace `withFile filePath ReadMode SBS.hGetContents` with `SBS.readFile filePath`. Don't know why we (I?) made this so complicated back then.

Reply:
Yes, fixed.